confmap/provider/googlesecretsprovider/provider.go

```diff
@@ -0,0 +1,78 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate mdatagen metadata.yaml
+
+package googlesecretsprovider // import "github.com/open-telemetry/opentelemetry-collector-contrib/confmap/provider/googlesecretsprovider"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	gax "github.com/googleapis/gax-go/v2"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"go.opentelemetry.io/collector/confmap"
+
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+)
+
+type secretsManagerClient interface {
+	AccessSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error)
+	Close() error
+}
+
+const (
+	schemeName = "googlesecretsprovider"
+)
+
+type provider struct {
+	client secretsManagerClient
+}
+
+func NewFactory() confmap.ProviderFactory {
+	return confmap.NewProviderFactory(newProvider)
+}
+
+func newProvider(confmap.ProviderSettings) confmap.Provider {
+	return &provider{client: nil}
+}
+
+func (p *provider) Retrieve(ctx context.Context, uri string, _ confmap.WatcherFunc) (*confmap.Retrieved, error) {
+	if !strings.HasPrefix(uri, schemeName+":") {
+		return nil, fmt.Errorf("%q uri is not supported by %q provider", uri, schemeName)
```

Comment:
This should wrap a sentinel error value. If we define a `var ErrSchemeNotSupported` then we can wrap it with `fmt.Errorf`. This allows the test to `require.ErrorIs` and check for `ErrSchemeNotSupported`.

Additionally, the error message should say `not supported by googlesecretsprovider`, not `schemeName` right? Otherwise it would be kind of a confusing message.

Reply:
I simply followed the conventions used in other existing providers:  https://github.com/open-telemetry/opentelemetry-collector/blob/c85ebbec779fa00c3fbe4a555eb0d420bdad7581/confmap/provider/fileprovider/provider.go#L47